presto-matching/src/main/java/com/facebook/presto/matching/Property.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.matching;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+public interface Property<F, T>
```

Comment:
It looks like an interface but it is not implemented anywhere. When I did a closer look it looks like a `PropertyPattern::Builder` (or `PropertyPattern::Provider`, `PropertyPattern::Description`). Maybe lets name that way and put into `PropertyPattern` class. What do you think? Have I missed something?

Then `PropertyPattern` could be renamed to just `Pattern` and `WithPattern` to `PropertyPattern`.

The value I see in current design is that the code like below looks nice and clean, but IMO does not express the meaning what this class represents.
```
Property<String, Integer> length = Property.property(String::length);
```